dnsclient.go

```diff
@@ -11,16 +11,27 @@ import (
 
 const hexDigit = "0123456789abcdef"
 
+var DefaultClient = &Client{resolver: net.DefaultResolver}
+
+type Client struct {
+	resolver *net.Resolver
```

Comment:
I believe we should export `resolver` and then remove `NewClient`. The `New` abstraction is a bit premature.

Reply:
If I do this, users of this lib will be able to create a Client without resolver, and this will lead to a crash.
So I would have to add a check on each function to ensure we have at least the default resolver.

Reply:
That panic still exists if you instantiated a `DNSClient` and ignored the New function.

If you're creating a `DNSClient` (or any foreign struct) you should probably glance at the fields.

Also, a New API here obscures the name of the fields (function calls are unkeyed) leading a more cognitive overhead when reviewing code, and makes the otherwise simple `DNSClient` type mysterious. My preference is to only introduce a New function when either:
1. The struct has complex initialization logic
2. Only some of the struct's fields are required (and it's not as simple as public = required, private = not)